maintenance_mode/http.py

```diff
@@ -93,7 +93,7 @@ def need_maintenance_response(request):
         return False
 
     try:
-        url_off = reverse('maintenance_mode_off')
+        url_off = reverse('maintenance_mode_off').replace(request.META['SCRIPT_NAME'], '')
```

Comment:
What happens if for example the `SCRIPT_NAME` value is a substring of `/maintenance-mode/off/`, eg. `off`? 
The resulting `url_off` value will be: `/maintenance-mode//`
I think that replacing it as you did is a little bit dangerous, maybe it solves your problem, but it could create new ones.

Reply:
Yes, you are right. Maybe the search should only look for the substrings placed at the front of the URL?

For example:
```python
url_off = url_off.replace(request.META.get('SCRIPT_NAME'), '') if url_off.startswith(request.META.get('SCRIPT_NAME')) else url_off
```

In that case, `SCRIPT_NAME = /off` does not eliminate the final `/off` in `/maintenance-mode/off/`. However, if the URL starts with a substring same as `SCRIPT_NAME`, then the problem is still here. I couldn't imagine how to get rid of this problem at the moment.

Reply:
- The script name could be moved to a variable as it is used twice
- The replace will replace all occurrences, this could be dangerous, we should limit only to first occurrence

Yes, the problem is still here but we reduced it only to extreme edge cases.